services/datamanager/builtin/sync/file_deletion.go

```diff
@@ -0,0 +1,260 @@
+	)
+	if err != nil && !shouldDelete {
+		logger.Warnf("Disk nearing capacity but data capture directory is below %f of that size, file deletion will not run",
+			CaptureDirToFSUsageRatio)
+	}
+	return shouldDelete, err
+}
+
+// returns false, nil if the threshold is not exceeded
+// returns false, error if an IO error was encountered
+// returns true, nil if the threshold is exceeded.
+func exceedsDeletionThreshold(
+	ctx context.Context,
+	captureDirPath string,
+	fileSystemSizeBytes float64,
+	captureDirToFSUsageRatio float64,
+) (bool, error) {
+	var dirSize float64
+	readSize := func(path string, d fs.DirEntry, err error) error {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				return nil
+			}
+			return err
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		fileInfo, err := d.Info()
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				return nil
+			}
+			return err
+		}
+		dirSize += float64(fileInfo.Size())
+		if dirSize/fileSystemSizeBytes >= captureDirToFSUsageRatio {
+			return errAtSizeThreshold
+		}
+		return nil
+	}
+
+	err := filepath.WalkDir(captureDirPath, readSize)
+	if err != nil {
+		if errors.Is(err, errAtSizeThreshold) {
+			return true, nil
+		}
+		return false, err
+	}
+	return false, nil
+}
+
+// deleteFiles temporarily public for tests.
+func deleteFiles(
+	ctx context.Context,
+	fileTracker *fileTracker,
+	deleteEveryNth int,
+	captureDirPath string,
+	logger logging.Logger,
+) (int, error) {
+	index := 0
+	deletedFileCount := 0
+	logger.Infof("Deleting every %dth file", deleteEveryNth)
+	fileDeletion := func(path string, d fs.DirEntry, err error) error {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		if err != nil {
+			// this can happen if after we start walking the dir, the file changes from .prog to .capture
+			// which throws a file not found error when we try to get the fileinfo. If we hit this, just
+			// swallow the error and continue walking
+			if errors.Is(err, fs.ErrNotExist) {
+				return nil
+			}
+			return err
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+		fileInfo, err := d.Info()
+		if err != nil {
+			// same reason as above
+			if errors.Is(err, fs.ErrNotExist) {
+				return nil
+			}
+			return err
+		}
+		isCompletedDataCaptureFile := strings.Contains(fileInfo.Name(), data.CompletedCaptureFileExt)
+		// if at nth file and the file is not currently being written, mark as in progress if possible
+		if isCompletedDataCaptureFile && index%deleteEveryNth == 0 {
+			if !fileTracker.markInProgress(path) {
+				logger.Debugw("Tried to mark file as in progress but lock already held", "file", d.Name())
+				return nil
+			}
+			if err := os.Remove(path); err != nil {
+				logger.Warnw("error deleting file", "error", err)
+				fileTracker.unmarkInProgress(path)
+				return err
+			}
+			logger.Infof("successfully deleted %s", d.Name())
+			deletedFileCount++
+		}
+		logger.Info("ignored: " + fileInfo.Name())
```

Comment:
I think this log line should be in an else

Reply:
removed